File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    cond do
      String.strip(input) == "" ->
        "Fine. Be that way!"

      String.match?(input, ~r/\?$/) ->
        "Sure."

      String.match?(input, ~r/[A-Z]{2,}|[А-Я]/) ->
        "Whoa, chill out!"

      true ->
        "Whatever."
    end
  end
end

File: ruby/bob/bob.rb
module Bob
  def self.hey(remark)
    if remark.strip.empty?
      'Fine. Be that way!'
    elsif remark.upcase == remark && remark.match(/[A-Z]+/)
      "Whoa, chill out!"
    elsif remark[-1] == "?"
      "Sure."
    else
      "Whatever."
    end
  end
end
